src/NumericalAlgorithms/Interpolation/InterpolationTarget.hpp

```diff
@@ -120,6 +120,256 @@ namespace intrp {
+///     call block_logical_coordinates and begin interpolating for the
+///     first temporal_id
+///   } else {
+///     Do nothing, because there is already a sequential interpolation in
+///     progress.
+///   }
+/// } else { // not sequential
+///    call block_logical_coordinates and begin interpolating for
+///    all newly-added TemporalIds.
+/// }
+///```
+///
+/// ###### New logic:
+///
+/// ```
+/// Add passed-in temporal_ids to PendingTemporalIds
+/// if (sequential) {
+///   if (TemporalIds is empty and
+///       PendingTemporalIds was empty before it was appended above) {
+///     call VerifyTemporalIdsAndSendPoints
+///   } else {
+///     Do nothing, because there is already a sequential interpolation in
+///     progress, or there is a callback already waiting.
+///   }
+/// } else { // not sequential
+///   if (TemporalIds is non-empty) {
+///     call block_logical_coordinates and begin interpolating for
+///     all TemporalIds.
+///   }
+///   if (PendingTemporalIds is non-empty) {
+///     call VerifyTemporalIdsAndSendPoints
+///   }
+///}
+///```
+///
+/// ##### Actions::InterpolationTargetReceiveVars
+///
+/// `Actions::InterpolationTargetReceiveVars` is called by the
+/// `Interpolator` when it is finished interpolating the current
+/// `temporal_id`.  For the sequential case, it needs to start
+/// interpolating for the next `temporal_id`.  The logic is, in pseudocode:
+///
+/// ###### Current logic:
+///
+///```
+/// if (sequential) {
+///   if (TemporalIds is not empty) {
+///     call block_logical_coordinates and interpolate for next temporal_id
+///   }
+/// }
+///```
+///
+/// ###### New logic:
+///
+///```
+/// if (sequential) {
+///   if (TemporalIds is not empty) {
+///     call block_logical_coordinates and interpolate for next temporal_id
+///   } else if (PendingTemporalIds is not empty) {
+///     call VerifyTemporalIdsAndSendPoints
+///   }
+/// }
+///```
+///
+/// ##### intrp::callbacks::FindApparentHorizon
+///
+/// `intrp::callbacks::FindApparentHorizon calls`
+/// `block_logical_coordinates` when it needs to start a new iteration
+/// of the horizon finder at the same `temporal_id`, so one might
+/// think you need to worry about up-to-date `FunctionOfTime`s. But
+/// since `intrp::callbacks::FindApparentHorizon` always works on the same
+/// `temporal_id` for which the `FunctionOfTime`s have already been
+/// verified as up-to-date from the last iteration, no special consideration
+/// of `FunctionOfTime`s need be done here.
+///
+/// #### Interpolation without using the Interpolator ParallelComponent
+///
+/// This case is easier than the case with the `Interpolator`, because
+/// the points are always time-independent in the frame
+/// `compute_target_points::frame`.
+///
+/// ##### Actions::EnsureFunctionOfTimeUpToDate
+///
+/// `Actions::EnsureFunctionOfTimeUpToDate` verifies that the
+///  `FunctionOfTime`s are up-to-date at the `DgElementArray`s current
+///  time.
```

Comment:
If I understand correctly this is the case handling observers.  With dense output the time used by an observer will not correspond exactly with the step time of the element.  I don't know if we want the observer events to have is_ready functions or just ensure events and similar always have all functions of time up to data or what, but I think it has to be handled more precisely than as a separate action.

Reply:
Agreed.  I like the idea of observer events having is_ready functions, because I think this will simplify the logic of non-observers (because they don't need to worry about the `FunctionOfTime`s being up to date for observers).

Note also that what I am calling `Actions::EnsureFunctionOfTimeUpToDate`, or something like it, will also be needed to take the time step, because the RHS will depend on the map.  So treatment of `Actions::EnsureFunctionOfTimeUpToDate` should be coordinated with what @osheamonn is doing for the control systems.

My plan is to do the with-`Interpolator` case first, since that is what is needed for the AH finder, and it should be independent of how `Element`s ensure that `FunctionOfTime`s are updated.